Makefile

```diff
@@ -70,6 +70,8 @@ export PATH := $(abspath $(TOOLS_BIN_DIR)):$(PATH)
 # Set --output-base for conversion-gen if we are not within GOPATH
 ifneq ($(abspath $(ROOT_DIR)),$(shell go env GOPATH)/src/sigs.k8s.io/cluster-api)
 	CONVERSION_GEN_OUTPUT_BASE := --output-base=$(ROOT_DIR)
+else
+	export GOPATH := $(shell go env GOPATH)
```

Comment:
Why is this needed?

Reply:
I ported it from CAPA. If you still have repos checked out in the traditional go path ways, then you end up with the `sigs.k8s.io` directories being generated. It's a fix up to https://github.com/kubernetes-sigs/cluster-api/pull/4349